packages/components/textarea/src/textarea.js

```diff
@@ -0,0 +1,116 @@
+import { TSElement, unsafeCSS, html, customElementDefineHelper } from '@tradeshift/elements';
+import css from './textarea.css';
+import { countLines, getRowsInRange } from './utils';
+
+/**
+ * @class TSTextArea
+ * @property {string} title - The title text for the header
+ * @property {boolean} disabled - Disables interaction with element
+ * @property {boolean} dir - Direction of the component 'rtl' or 'ltr'
+ * @private {HTMLTextAreaElement} textarea - Textarea outside of shadow DOM to support forms
+ */
+class TSTextArea extends TSElement {
+	static get styles() {
+		return [TSElement.styles, unsafeCSS(css)];
+	}
+
+	static get properties() {
+		return {
+			title: { type: String, reflect: true },
+			disabled: { type: Boolean, reflect: true },
+			value: { type: String, reflect: true },
+			dir: { type: Boolean, reflect: true }
+		};
+	}
+
+	get direction() {
+		return this.dir ? this.dir : this.bodyDir;
+	}
+
+	constructor() {
+		super();
+		this.textarea = null;
+		this.updateChangedHandler = this.updateChangedHandler.bind(this);
+	}
+
+	attributeChangedCallback(name, oldVal, newVal) {
+		super.attributeChangedCallback(name, oldVal, newVal);
+		if (!this.textarea) {
+			return;
+		}
+
+		switch (name) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add a "default" clause to this "switch" statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.ts.sv/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)